src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -1653,6 +1656,31 @@ void testVersionReplacedMapReducedDuringPromotionToExtensionBranch() throws Serv
 		assertEquals(0, branchService.findBranchOrThrow("MAIN/SNOMEDCT-DK").getVersionsReplacedCounts().get("Concept").intValue());
 	}
 
+	@Test
+	void testPromotionAbortedIfBranchReviewIncomplete() throws ServiceException, InterruptedException {
+		assertBranchState("MAIN", Branch.BranchState.UP_TO_DATE);
+		assertBranchState("MAIN/A", Branch.BranchState.UP_TO_DATE);
+		assertBranchState("MAIN/A/A1", Branch.BranchState.UP_TO_DATE);
+		givenBranchReviewIncomplete();
+
+		// Create concept on A1
+		final String conceptId = "10000123";
+		conceptService.create(new Concept(conceptId), "MAIN/A/A1");
+		assertBranchStateAndConceptVisibility("MAIN", Branch.BranchState.UP_TO_DATE, conceptId, false);
+		assertBranchStateAndConceptVisibility("MAIN/A", Branch.BranchState.UP_TO_DATE, conceptId, false);
+		assertBranchStateAndConceptVisibility("MAIN/A/A1", Branch.BranchState.FORWARD, conceptId, true);
+
+		// Promote
+		MergeRequest mergeRequest = new MergeRequest("MAIN/A/A1", "MAIN/A", null, null);
+		BranchMergeJob branchMergeJob = branchMergeService.mergeBranchAsync(mergeRequest);
+
+		// Promotion should have failed
+		Thread.sleep(5000);
```

Comment:
The sleep statement could be avoided by using the `BranchMergeService.mergeBranchSync()` method.

Reply:
Thanks, that'll help with reducing the test time! I have made that change.